internal/pkg/telemetry/windows_cpu.go

```diff
@@ -0,0 +1,84 @@
+// +build windows
+
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package telemetry
+
+import (
+	"math"
+	"syscall"
+	"unsafe"
+
+	"bitbucket.org/bertimus9/systemstat"
+)
+
+var (
+	modkernel32        = syscall.NewLazyDLL("kernel32.dll")
+	procGetSystemTimes = modkernel32.NewProc("GetSystemTimes")
+)
+
+func PollCpu() (cpuSnapshot CpuUsage) {
+	var idle, kernel, user FILETIME
+
+	GetSystemTimes(&idle, &kernel, &user)
+	idleFirst := idle.DwLowDateTime | (idle.DwHighDateTime << 32)
+	kernelFirst := kernel.DwLowDateTime | (kernel.DwHighDateTime << 32)
+	userFirst := user.DwLowDateTime | (user.DwHighDateTime << 32)
+
+	return CpuUsage{
+		Busy:  uint64(userFirst),               // linuxSample.Nice + linuxSample.User,
+		Idle:  uint64(idleFirst),               //linuxSample.Idle,
+		Total: uint64(kernelFirst + userFirst), // linuxSample.Total,
+	}
+
+}
+
+func AvgCpuUsage(init, final CpuUsage) (avg float64) {
+	// SimpleAverage only uses idle and total, so only copy those
+	windowsInit := systemstat.CPUSample{
+		Idle:  init.Idle,
+		Total: init.Total,
+	}
+
+	windowsFinal := systemstat.CPUSample{
+		Idle:  final.Idle,
+		Total: final.Total,
+	}
+
+	avg = systemstat.GetSimpleCPUAverage(windowsInit, windowsFinal).BusyPct
+
+	if avg < .000001 || math.IsNaN(avg) {
+		return 0.0
+	}
+
+	return avg
+}
+
+func GetSystemTimes(lpIdleTime, lpKernelTime, lpUserTime *FILETIME) bool {
+	ret, _, _ := procGetSystemTimes.Call(
+		uintptr(unsafe.Pointer(lpIdleTime)),
+		uintptr(unsafe.Pointer(lpKernelTime)),
+		uintptr(unsafe.Pointer(lpUserTime)))
+
+	return ret != 0
+}
+
+// FILETIME Struct from: http://msdn.microsoft.com/en-us/library/windows/desktop/ms724284.aspx
+type FILETIME struct {
+	DwLowDateTime  uint32

```

Comment:
Using Hungarian notation doesn't seem like the answer in Go.

Reply:
No requirement that I'm aware of, just did it to match the Windows API so it was easy to follow what was what.